### DISCUSSION TITLE
Building Custom Ubuntu 24.04 GitHub Runner Image with Packer

### DISCUSSION THREAD

**Discussion by koss822** [community/community#160650] [Actions] [answered]
Hello Community,

I am currently running MicroK8s on an Ubuntu 24.04 server, where I host self-managed GitHub runners. However, I am struggling to find existing Packer configurations or resources that would help me create a custom GitHub runner image specifically on Ubuntu 24.04 server. The only resources I have found so far are a PowerShell script for Azure deployment, which doesn't meet my requirements.

Could anyone guide me on how to create a Packer template for building a custom Ubuntu 24.04 GitHub runner image? Alternatively, if there are any alternative methods or resources available for achieving this, I would greatly appreciate your insights.

Thank you!

**Comment by RaviAkshintala** [comment, marked as answer]
Hi @koss822 Could you please try with the below template.
```
variable "github_runner_version" {
  default = "latest"
}

# Builder configuration
builder "docker" {
  image = "ubuntu:24.04"
  commit = true
}

# Provisioner to install dependencies and GitHub runner
provisioner "shell" {
  inline = [
    "apt-get update",
    "apt-get install -y curl",
    "curl -o actions-runner-linux-x64.tar.gz -L https://github.com/actions/runner/releases/download/v${var.github_runner_version}/actions-runner-linux-x64-${var.github_runner_version}.tar.gz",
    "tar xzf ./actions-runner-linux-x64.tar.gz",
    "./config.sh --url https://github.com/your-repo --token YOUR_GITHUB_TOKEN",
    "sudo ./svc.sh install",
    "sudo ./svc.sh start"
  ]
}

# Optional: Output the image ID
output "image_id" {
  value = "${builder.image}"
}
```
Run the following command to build your image:
`packer build github-runner.pkr.hcl`

**Comment by koss822** [comment]
1. I will use image for multiple repos so not sure what this is for
`"./config.sh --url https://github.com/your-repo --token YOUR_GITHUB_TOKEN",`
2. I need to upload image to specific repository url, not sure how to define this
